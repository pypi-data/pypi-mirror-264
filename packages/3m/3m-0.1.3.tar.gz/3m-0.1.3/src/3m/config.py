import httpx
from llama_index.core import ServiceContext
from llama_index.llms.openai_like import OpenAILike
from llama_index.embeddings.openai import OpenAIEmbedding
from llama_index.core.response_synthesizers import ResponseMode
from llama_index.core.query_engine import RetrieverQueryEngine
LLM = OpenAILike(
            model="internlm/internlm2-chat-20b",
            api_key='sd-66C45asdas8',
            is_chat_model=True,
            api_base='http://103.177.28.196:8003/v1',
            http_client=httpx.Client(trust_env=True),
            max_tokens=2000,
            additional_kwargs={"stop":["<|im_end|>"]}
        )
        
EMBEDDING=OpenAIEmbedding(api_base='http://103.177.28.196:8003/v1',api_key='sd-66C45asdas8',embed_batch_size=2)