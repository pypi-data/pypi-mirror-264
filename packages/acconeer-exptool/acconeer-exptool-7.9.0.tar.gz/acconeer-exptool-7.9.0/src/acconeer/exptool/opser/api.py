# Copyright (c) Acconeer AB, 2023
# All rights reserved
from __future__ import annotations

import typing as t

import h5py

from . import core
from .registry_persistor import RegistryPersistor


_T = t.TypeVar("_T")


def serialize(
    instance: t.Any, group: h5py.Group, *, override_type: t.Optional[core.TypeLike] = None
) -> None:
    """
    Serialize and save an arbitrary object to the specified group
    """
    type_tree = core.create_type_tree(override_type or type(instance))
    core.sanitize_instance(instance, type_tree)
    RegistryPersistor(group, "/", type_tree).save(instance)


def deserialize(group: h5py.Group, typ: t.Type[_T]) -> _T:
    """
    Try to load and deserialize an object of type 'typ' from the specified group
    """
    type_tree = core.create_type_tree(typ)
    loaded = RegistryPersistor(group, "/", type_tree).load()
    core.sanitize_instance(loaded, type_tree)

    return loaded  # type: ignore[no-any-return]


register_persistor = RegistryPersistor.register_persistor


def register_json_presentable(__class: t.Type[core.JsonPresentable]) -> None:
    """
    Register a JSON presentable class (class with 'to_json' & 'from_json')

    Instances of this type will be saved as a JSON string.

    This can be called many times with the same type without repercussions.
    """
    RegistryPersistor.register_persistor(core.Persistor.from_json_presentable(__class))
