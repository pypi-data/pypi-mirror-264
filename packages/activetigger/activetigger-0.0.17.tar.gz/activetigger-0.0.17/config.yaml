path: ./projects
secret_key: f63aeb7426d2c8a3defc02a3e788c2f311482d6cff557c2c5bdebc71d67b507a #HS256
