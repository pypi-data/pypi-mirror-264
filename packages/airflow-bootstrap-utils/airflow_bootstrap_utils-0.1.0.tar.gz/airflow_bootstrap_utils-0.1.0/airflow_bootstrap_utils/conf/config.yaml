---
# Attribute: default_operator_type
# Note: This will be the default Airflow
# operator type that will be used for all
# tasks in the generated Airflow DAG Python
# script file.
default_operator_type: bash
# Attribute: default_dag_template_name
# Note: This the default template name for the jinja2
# template file that will be used to generate the
# Airflow DAG Python script file.
default_dag_template_name: dag_template.py
