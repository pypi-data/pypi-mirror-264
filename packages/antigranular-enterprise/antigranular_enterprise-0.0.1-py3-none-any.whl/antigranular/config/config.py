import os
import configparser
import yaml

class Config:
    def __init__(self):
        home_dir = os.path.expanduser('~')
        config_dir = os.path.join(home_dir, '.agent')
        config_path = os.path.join(config_dir, 'config')
        self.config_path = config_path
        self.config = configparser.ConfigParser()
        self.profile = 'DEFAULT'
        self.AG_EXEC_TIMEOUT = 1000

        if not os.path.exists(self.config_path):
            os.makedirs(config_dir, exist_ok=True)
        else:
            self.read_config(self.profile)

    def read_config(self, profile):
        try:
            self.profile = profile
            self.config.read(self.config_path)
            self._load_config_values()
        except Exception as e:
            print(f"An error occurred while reading the config: {e}")

    def write_config(self, yaml_config, profile):
        try:
            self.profile = profile
            config_dict = yaml.safe_load(yaml_config)
            self.config[self.profile] = config_dict
            with open(self.config_path, 'w') as config_file:
                self.config.write(config_file)
            self._load_config_values()
        except Exception as e:
            print(f"An error occurred while writing the config: {e}")


    def _write_config(self):
        with open(self.config_path, 'w') as config_file:
            self.config.write(config_file)

    def _load_config_values(self):
        for key, value in self.config[self.profile].items():
            setattr(self, key.upper(), value)
        for section in self.config.sections():
            section_dict = getattr(self, section.upper(), {})
            for key, value in self.config[section].items():
                section_dict[key.upper()] = value
            setattr(self, section.upper(), section_dict)