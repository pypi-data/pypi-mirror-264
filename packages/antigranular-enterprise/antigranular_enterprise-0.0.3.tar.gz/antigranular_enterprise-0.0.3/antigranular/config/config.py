import os
import configparser
import yaml

class Config:
    """
    Represents a configuration object that handles reading and writing configuration files.

    Attributes:
        config_path (str): The path to the configuration file.
        config (ConfigParser): The configuration parser object.
        profile (str): The current profile being used.
        AG_EXEC_TIMEOUT (int): The execution timeout value.

    Methods:
        __init__(): Initializes a new instance of the Config class.
        __getattr__(name): Gets called when an attribute with the given name is not found.
        load_config(): Loads the configuration from the file.
        read_config(profile): Reads the configuration for the specified profile.
        write_config(yaml_config, profile): Writes the configuration to the file for the specified profile.
        _write_config(): Writes the configuration to the file.
        _load_config_values(): Loads the configuration values into the class attributes.
    """

    def __init__(self):
        """
        Initializes a new instance of the Config class.

        The configuration file is located in the user's home directory under the '.agent' folder.
        The default profile is set to 'DEFAULT'.
        The default execution timeout is set to 1000 milliseconds.
        """
        home_dir = os.path.expanduser('~')
        config_dir = os.path.join(home_dir, '.agent')
        config_path = os.path.join(config_dir, 'config')
        self.config_path = config_path
        self.config = configparser.ConfigParser()
        self.profile = 'DEFAULT'
        self.AG_EXEC_TIMEOUT = 1000

        self.load_config()
    
    def __getattr__(self, name):
        """
        Gets called when an attribute with the given name is not found.

        Args:
            name (str): The name of the attribute.

        Raises:
            AttributeError: If the attribute is not found in the configuration file.

        Returns:
            None
        """
        raise AttributeError(f"Please update the config file {self.config_path} profile {self.profile} with the attribute {name}")

    def load_config(self):
        """
        Loads the configuration from the file.

        If the configuration file does not exist, a message is printed.
        Otherwise, the configuration is read for the current profile.
        """
        if not os.path.exists(self.config_path):
            print(f"Config file not found at {self.config_path} Please use the 'write_config' method to create a new config.")
        else:
            self.read_config(self.profile)

    def read_config(self, profile):
        """
        Reads the configuration for the specified profile.

        Args:
            profile (str): The profile to read the configuration for.

        Raises:
            Exception: If an error occurs while reading the configuration.

        Returns:
            None
        """
        try:
            self.profile = profile
            self.config.read(self.config_path)
            self._load_config_values()
        except Exception as e:
            print(f"An error occurred while reading the config: {e}")

    def write_config(self, yaml_config, profile):
        """
        Writes the configuration to the file for the specified profile.

        Args:
            yaml_config (str): The YAML configuration string.
            profile (str): The profile to write the configuration for.

        Raises:
            Exception: If an error occurs while writing the configuration.

        Returns:
            None
        """
        try:
            self.profile = profile
            config_dict = yaml.safe_load(yaml_config)
            self.config[self.profile] = config_dict
            with open(self.config_path, 'w') as config_file:
                self.config.write(config_file)
            self._load_config_values()
        except Exception as e:
            print(f"An error occurred while writing the config: {e}")


    def _write_config(self):
        """
        Writes the configuration to the file.

        This method is called internally when writing the configuration.
        """
        with open(self.config_path, 'w') as config_file:
            self.config.write(config_file)

    def _load_config_values(self):
        """
        Loads the configuration values into the class attributes.

        This method is called internally when reading the configuration.
        """
        for key, value in self.config[self.profile].items():
            setattr(self, key.upper(), value)
        for section in self.config.sections():
            section_dict = getattr(self, section.upper(), {})
            for key, value in self.config[section].items():
                section_dict[key.upper()] = value
            setattr(self, section.upper(), section_dict)

config = Config()