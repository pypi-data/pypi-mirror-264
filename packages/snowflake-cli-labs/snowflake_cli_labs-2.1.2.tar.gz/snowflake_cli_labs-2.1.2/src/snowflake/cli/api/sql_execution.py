from __future__ import annotations

import logging
from contextlib import contextmanager
from functools import cached_property
from io import StringIO
from textwrap import dedent
from typing import Iterable, Optional

from click import ClickException
from snowflake.cli.api.cli_global_context import cli_context
from snowflake.cli.api.constants import ObjectType
from snowflake.cli.api.exceptions import (
    DatabaseNotProvidedError,
    SchemaNotProvidedError,
    SnowflakeSQLExecutionError,
)
from snowflake.cli.api.project.util import (
    identifier_to_show_like_pattern,
    unquote_identifier,
)
from snowflake.cli.api.utils.cursor import find_first_row
from snowflake.connector.cursor import DictCursor, SnowflakeCursor
from snowflake.connector.errors import ProgrammingError


class SqlExecutionMixin:
    def __init__(self):
        pass

    @property
    def _conn(self):
        return cli_context.connection

    @cached_property
    def _log(self):
        return logging.getLogger(__name__)

    def _execute_string(
        self,
        sql_text: str,
        remove_comments: bool = False,
        return_cursors: bool = True,
        cursor_class: SnowflakeCursor = SnowflakeCursor,
        **kwargs,
    ) -> Iterable[SnowflakeCursor]:
        """
        This is a custom implementation of SnowflakeConnection.execute_string that returns generator
        instead of list. In case of executing multiple queries are executed one by one. This mean we can
        access result of previous queries while evaluating next one. For example, we can print the results.
        """
        self._log.debug("Executing %s", sql_text)
        stream = StringIO(sql_text)
        stream_generator = self._conn.execute_stream(
            stream, remove_comments=remove_comments, cursor_class=cursor_class, **kwargs
        )
        return stream_generator if return_cursors else list()

    def _execute_query(self, query: str, **kwargs):
        *_, last_result = self._execute_queries(query, **kwargs)
        return last_result

    def _execute_queries(self, queries: str, **kwargs):
        return list(self._execute_string(dedent(queries), **kwargs))

    def use(self, object_type: ObjectType, name: str):
        return self._execute_query(f"use {object_type.value.sf_name} {name}")

    @contextmanager
    def use_role(self, new_role: str):
        """
        Switches to a different role for a while, then switches back.
        This is a no-op if the requested role is already active.
        """
        role_result = self._execute_query(
            f"select current_role()", cursor_class=DictCursor
        ).fetchone()
        prev_role = role_result["CURRENT_ROLE()"]
        is_different_role = new_role.lower() != prev_role.lower()
        if is_different_role:
            self._log.debug("Assuming different role: %s", new_role)
            self._execute_query(f"use role {new_role}")
        try:
            yield
        finally:
            if is_different_role:
                self._execute_query(f"use role {prev_role}")

    def _execute_schema_query(self, query: str, **kwargs):
        self.check_database_and_schema()
        return self._execute_query(query, **kwargs)

    def check_database_and_schema(self) -> None:
        """
        Checks if the connection database and schema are set and that they actually exist in Snowflake.
        """
        self.check_schema_exists(self._conn.database, self._conn.schema)

    def check_database_exists(self, database: str) -> None:
        """
        Checks that database is provided and that it is a valid database in
        Snowflake. Note that this could fail for a variety of reasons,
        including not authorized to use database, database doesn't exist,
        database is not a valid identifier, and more.
        """
        if not database:
            raise DatabaseNotProvidedError()
        try:
            self._execute_query(f"USE DATABASE {database}")
        except ProgrammingError as e:
            raise ClickException(f"Exception occurred: {e}.") from e

    def check_schema_exists(self, database: str, schema: str) -> None:
        """
        Checks that schema is provided and that it is a valid schema in Snowflake.
        Note that this could fail for a variety of reasons,
        including not authorized to use schema, schema doesn't exist,
        schema is not a valid identifier, and more.
        """
        self.check_database_exists(database)
        if not schema:
            raise SchemaNotProvidedError()
        try:
            self._execute_query(f"USE {database}.{schema}")
        except ProgrammingError as e:
            raise ClickException(f"Exception occurred: {e}.") from e

    def to_fully_qualified_name(
        self, name: str, database: Optional[str] = None, schema: Optional[str] = None
    ):
        current_parts = name.split(".")
        if len(current_parts) == 3:
            # already fully qualified name
            return name.upper()

        if not database:
            if not self._conn.database:
                raise ClickException(
                    "Default database not specified in connection details."
                )
            database = self._conn.database

        if len(current_parts) == 2:
            # we assume name is in form of `schema.object`
            return f"{database}.{name}".upper()

        schema = schema or self._conn.schema or "public"
        database = database or self._conn.database
        return f"{database}.{schema}.{name}".upper()

    def show_specific_object(
        self,
        object_type_plural: str,
        unqualified_name: str,
        name_col: str = "name",
        in_clause: str = "",
        check_schema: bool = False,
    ) -> Optional[dict]:
        """
        Executes a "show <objects> like" query for a particular entity with a
        given (unqualified) name. This command is useful when the corresponding
        "describe <object>" query does not provide the information you seek.
        """
        if check_schema:
            self.check_database_and_schema()
        show_obj_query = f"show {object_type_plural} like {identifier_to_show_like_pattern(unqualified_name)} {in_clause}".strip()
        show_obj_cursor = self._execute_query(  # type: ignore
            show_obj_query, cursor_class=DictCursor
        )
        if show_obj_cursor.rowcount is None:
            raise SnowflakeSQLExecutionError(show_obj_query)
        show_obj_row = find_first_row(
            show_obj_cursor,
            lambda row: row[name_col] == unquote_identifier(unqualified_name),
        )
        return show_obj_row
