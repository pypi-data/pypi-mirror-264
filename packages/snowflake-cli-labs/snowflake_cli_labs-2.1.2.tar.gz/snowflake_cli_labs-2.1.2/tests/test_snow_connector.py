from tests.testing_utils.fixtures import *


# Used as a solution to syrupy having some problems with comparing multilines string
class CustomStr(str):
    def __repr__(self):
        return str(self)


MOCK_CONNECTION = {
    "database": "databaseValue",
    "schema": "schemaValue",
    "role": "roleValue",
    "show": "warehouseValue",
}


@pytest.mark.parametrize(
    "cmd,expected",
    [
        ("snow sql", "SNOWCLI.SQL"),
        ("snow show warehouses", "SNOWCLI.SHOW.WAREHOUSES"),
    ],
)
@mock.patch("snowflake.connector.connect")
@mock.patch("snowflake.cli.app.snow_connector.command_info")
def test_command_context_is_passed_to_snowflake_connection(
    mock_command_info, mock_connect, cmd, expected, test_snowcli_config
):
    from snowflake.cli.app.snow_connector import connect_to_snowflake
    from snowflake.cli.api.config import config_init

    config_init(test_snowcli_config)

    mock_ctx = mock.Mock()
    mock_ctx.command_path = cmd
    mock_command_info.return_value = expected

    connect_to_snowflake()

    mock_connect.assert_called_once_with(
        application=expected,
        database="db_for_test",
        schema="test_public",
        role="test_role",
        warehouse="xs",
        password="dummy_password",
    )


@mock.patch.dict(os.environ, {}, clear=True)
def test_returns_nice_error_in_case_of_connectivity_error(runner):
    result = runner.invoke(["sql", "-q", "select 1"])
    assert result.exit_code == 1
    assert "Invalid connection configuration" in result.output
    assert "User is empty" in result.output


@mock.patch("snowflake.connector")
def test_no_output_from_connection(mock_connect, runner):
    funny_text = "what's taters, my precious?"

    def _mock(*args, **kwargs):
        print(funny_text)
        return mock.MagicMock()

    mock_connect.connect = _mock

    result = runner.invoke(["sql", "-q", "select 1"])
    assert funny_text not in result.output
