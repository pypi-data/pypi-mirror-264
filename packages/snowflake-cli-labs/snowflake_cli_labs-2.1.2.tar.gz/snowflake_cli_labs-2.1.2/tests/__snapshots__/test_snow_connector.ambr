# serializer version: 1
# name: test_upload_file_to_stage[namedStageValue-False]
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  
  
  
  put file://file_pathValue @namedStageValuepathValue auto_compress=false parallel=4 overwrite=overwriteValue;
  '''
# ---
# name: test_upload_file_to_stage[namedStageValue-True]
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  
  
  create stage if not exists namedStageValue comment='deployments managed by Snowflake CLI';
  
  
  put file://file_pathValue @namedStageValuepathValue auto_compress=false parallel=4 overwrite=overwriteValue;
  '''
# ---
# name: test_upload_file_to_stage[snow://embeddedStageValue-False]
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  
  
  
  put file://file_pathValue snow://embeddedStageValuepathValue auto_compress=false parallel=4 overwrite=overwriteValue;
  '''
# ---
# name: test_upload_file_to_stage[snow://embeddedStageValue-True]
  '''
  use role roleValue;
  use warehouse warehouseValue;
  use database databaseValue;
  use schema schemaValue;
  
  
  create stage if not exists snow://embeddedStageValue comment='deployments managed by Snowflake CLI';
  
  
  put file://file_pathValue snow://embeddedStageValuepathValue auto_compress=false parallel=4 overwrite=overwriteValue;
  '''
# ---
